packages/calcite-components/src/components.d.ts

```diff
@@ -10079,6 +10080,10 @@ declare namespace LocalJSX {
           * Fires when the close button is clicked.
          */
         "onCalciteListItemClose"?: (event: CalciteListItemCustomEvent<void>) => void;
+        /**
+          * Fires when the open button is clicked.
+         */
+        "onCalciteListItemOpen"?: (event: CalciteListItemCustomEvent<void>) => void;
```

Comment:
Mostly because we already have ‘close’ event for the “close button” interaction, but is it possible to use Toggle here or something besides ‘open’? We use ‘calcitePanelToggle‘ in Panel. It could also fire when collapsed?

Reply:
Yeah I can do that. its mapped to the open property though.

Reply:
Yeah, it’s tough with existing ‘close’ as that, in practice, is more of “dismiss” - is Toggle is too close to ‘select’ event?  OpenToggle?

Reply:
Updated. Ill let @jcfranco comment on name as well.

Reply:
**TL;DR**: I think we should go with `calciteListItemToggle` event firing when opened/closed. 

I'd like to avoid introducing an event name that mixes both open/close and toggling. We could introduce an individual toggle event (similar to `block`'s toggle event before we added open/close events) and check the event target's `open` value for now until we look more into the following:

* revisit close/dismiss/remove type props and their respective events (e.g., `tip` has a [closed](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/components/tip/tip.tsx#L50) prop and [dismissed](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/components/tip/tip.tsx#L141) event, `flow-item` has a [closed](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/components/flow-item/flow-item.tsx#L198) prop and a [close](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/components/flow-item/flow-item.tsx#L198) event, `pick-list-item` [deprecated] has no `close` prop, but similar affordance and a [removed](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/components/pick-list-item/pick-list-item.tsx#L244) event).
* use `open`/`close`, `expand`/`collapse` per https://github.com/Esri/calcite-design-system/issues/6473#issuecomment-1730103153.

Unfortunately, `list` is in a bit of a tough spot with some of these props and patterns we're adopting.

Reply:
Thanks 👍